### PR TITLE
feat(components): Add PasswordInput component

### PR DESCRIPTION
**Pull request by burakguneli**

### CODE REVIEW

File: src/form/password-input/PasswordInput.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  onChange,
+  hasError,
+  placeholder,
+  customClassName,
+  hidePasswordButtonIcon,
+  showPasswordButtonIcon,
+  passwordVisibilityButtonTestId
+}: PasswordInputProps) {
+  const [isPasswordShown, setPasswordVisibility] = useState(false);
+  const passwrodInputIconClassName = classNames("password-input__icon", {
```
Comment (jamcry) [inline src/form/password-input/PasswordInput.tsx:40]:
```suggestion
  const passwordInputIconClassName = classNames("password-input__icon", {
```
There is a small typo here

File: src/form/password-input/PasswordInput.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  testid,
+  name,
+  value,
+  onChange,
+  hasError,
+  placeholder,
+  customClassName,
+  hidePasswordButtonIcon,
+  showPasswordButtonIcon,
+  passwordVisibilityButtonTestId
```
Comment (jamcry) [inline src/form/password-input/PasswordInput.tsx:37]: I think this we can generate this inside the component using `testid`, and remove this prop
```
passwordVisibilityButtonTestId = `${testid}-password-visibility-icon`;
```

File: src/form/password-input/_password-input.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    position: absolute;
+    top: 0;
+    right: 0;
+
+    height: 48px;
+  }
+
+  .password-input__icon--is-visible {
+    display: inline-block;
+  }
```
Comment (jamcry) [inline src/form/password-input/_password-input.scss:14]: I guess we don't need to nest this

File: src/form/password-input/PasswordInput.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+export interface PasswordInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  onChange: React.ReactEventHandler<HTMLInputElement>;
+  onFocus?: React.ReactEventHandler<HTMLInputElement>;
+  onBlur?: React.ReactEventHandler<HTMLInputElement>;
+  onKeyDown?: React.KeyboardEventHandler<HTMLInputElement>;
+  onKeyUp?: React.KeyboardEventHandler<HTMLInputElement>;
+  onInput?: React.KeyboardEventHandler<HTMLInputElement>;
+  passwordVisibilityButtonTestId?: string;
+  hidePasswordButtonIcon?: Node;
```
Comment (mucahit) [inline src/form/password-input/PasswordInput.tsx:17]:
```suggestion
  hideIcon?: React.ReactNode;
  revealIcon?: React.ReactNode
```

File: src/form/password-input/PasswordInput.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  showPasswordButtonIcon?: Node;
+  testid: string;
```
Comment (mucahit) [inline src/form/password-input/PasswordInput.tsx:19]:
```suggestion
  testid?: string;
```

File: src/form/password-input/_password-input.scss [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+  .input-container-right-icon {
+    position: absolute;
+    top: 0;
+    right: 0;
+
+    height: 48px;
+  }
+
+  .password-input__icon--is-visible {
```
Comment (mucahit) [inline src/form/password-input/_password-input.scss:12]:
```suggestion
  &__icon--is-visible {
```

File: src/form/password-input/PasswordInput.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import "./_password-input.scss";
+
+import React, {useState} from "react";
+import classNames from "classnames";
+
+import Input from "../input/Input";
+import Button from "../../button/Button";
+
+export interface PasswordInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
```
Comment (mucahit) [inline src/form/password-input/PasswordInput.tsx:9]: How is this one?

```suggestion
export interface PasswordInputProps extends Omit<InputProps, "leftIcon" | "rightIcon"> {
  hideIcon?: React.ReactNode;
  revealIcon?: React.ReactNode
}
```

**Review by mucahit — CHANGES_REQUESTED** [review]
Thanks! @burakguneli 

Here are a few minor requests

**Review by bilgeyucel — APPROVED** [review]

**Review by edizcelik — APPROVED** [review]

**Review by jamcry — APPROVED** [review]

File: src/form/password-input/PasswordInput.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
+import "./_password-input.scss";
+
+import React, {useState} from "react";
+import classNames from "classnames";
+
+import Input, {InputProps} from "../input/Input";
+import Button from "../../button/Button";
+
+export interface PasswordInputProps extends Omit<InputProps, "leftIcon" | "rightIcon"> {
```
Comment (mucahit) [inline src/form/password-input/PasswordInput.tsx:9]: Let's convert to this, as far as I can see you're not overriding `InputProps` in here 

```suggestion
export interface PasswordInputProps extends Omit<InputProps, "leftIcon" | "rightIcon"> {
  hideIcon?: React.ReactNode;
  revealIcon?: React.ReactNode;
}
```

**Review by mucahit — APPROVED** [review]

**Pull request merged**